prow/cmd/hmac/main.go

```diff
@@ -0,0 +1,389 @@
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/yaml"
+
+	"k8s.io/test-infra/pkg/flagutil"
+	"k8s.io/test-infra/prow/cmd/hmac/updater"
+	"k8s.io/test-infra/prow/config"
+	prowflagutil "k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/logrusutil"
+)
+
+type options struct {
+	configPath string
+
+	dryRun     bool
+	github     prowflagutil.GitHubOptions
+	kubernetes prowflagutil.KubernetesOptions
+
+	hookUrl                  string
+	hmacTokenSecretNamespace string
+	hmacTokenSecretName      string
+	hmacTokenKey             string
+}
+
+func (o *options) validate() error {
+	for _, group := range []flagutil.OptionGroup{&o.kubernetes, &o.github} {
+		if err := group.Validate(o.dryRun); err != nil {
+			return err
+		}
+	}
+
+	if o.configPath == "" {
+		return errors.New("required flag --config-path was unset")
+	}
+	if o.hookUrl == "" {
+		return errors.New("required flag --hook-url was unset")
+	}
+	if o.hmacTokenSecretName == "" {
+		return errors.New("required flag --hmac-token-secret-name was unset")
+	}
+	if o.hmacTokenKey == "" {
+		return errors.New("required flag --hmac-token-key was unset")
+	}
+
+	return nil
+}
+
+func gatherOptions(fs *flag.FlagSet, args ...string) options {
+	var o options
+
+	o.github.AddFlags(fs)
+	o.kubernetes.AddFlags(fs)
+
+	fs.StringVar(&o.configPath, "config-path", "", "Path to config.yaml.")
+	fs.BoolVar(&o.dryRun, "dry-run", true, "Dry run for testing. Uses API tokens but does not mutate.")
+
+	fs.StringVar(&o.hookUrl, "hook-url", "", "Prow hook external webhook URL (e.g. https://prow.k8s.io/hook).")
+	fs.StringVar(&o.hmacTokenSecretNamespace, "hmac-token-secret-namespace", "default", "Name of the namespace on the cluster where the hmac-token secret is in.")
+	fs.StringVar(&o.hmacTokenSecretName, "hmac-token-secret-name", "", "Name of the secret on the cluster containing the GitHub HMAC secret.")
+	fs.StringVar(&o.hmacTokenKey, "hmac-token-key", "", "Key of the hmac token in the secret.")
+	fs.Parse(args)
+	return o
+}
+
+type client struct {
+	options options
+
+	kubernetesClient kubernetes.Interface
+
+	currentHMACMap map[string]github.HmacsForRepo
+	newHMACConfig  config.ManagedWebhooks
+
+	hmacMapForBatchUpdate map[string]string
+}
+
+func main() {
+	logrusutil.ComponentInit()
+
+	o := gatherOptions(flag.NewFlagSet(os.Args[0], flag.ExitOnError), os.Args[1:]...)
+	if err := o.validate(); err != nil {
+		logrus.WithError(err).Fatal("Invalid options")
+	}
+
+	kc, err := o.kubernetes.InfrastructureClusterClient(o.dryRun)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error creating Kubernetes client for infrastructure cluster.")
+	}
+
+	currentHmacYaml, err := getCurrentHmacTokens(kc, o.hmacTokenSecretNamespace, o.hmacTokenSecretName, o.hmacTokenKey)
+	if err != nil {
+		logrus.WithError(err).Fatal("Error getting the current hmac yaml.")
+	}
+
+	currentHMACMap := map[string]github.HmacsForRepo{}
```

Comment:
There is no need to instantiate a map that you then immediately delete in the next line.

This should be `var currentHMACMap map[string]github.HmacsForRepo`

Reply:
Done.